tests/test_cargo_test.rs

```diff
@@ -46,6 +46,41 @@ test result: ok. 1 passed; 0 failed; 0 ignored; 0 measured
         RUNNING)));
 });
 
+test!(cargo_test_release {
+    let p = project("foo")
+        .file("Cargo.toml", basic_bin_manifest("foo").as_slice())
+        .file("src/foo.rs", r#"
+            fn hello() -> &'static str {
+                "hello"
+            }
+
+            pub fn main() {
+                println!("{}", hello())
+            }
+
+            #[test]
+            fn test_hello() {
+                assert_eq!(hello(), "hello")
+            }"#);
+
+    assert_that(process(&p.bin("foo")).unwrap(),
+                execs().with_stdout("hello\n"));
+
+    assert_that(p.cargo("test").arg("-v").arg("--release"),
+                execs().with_stdout(format!("\
+{} foo v0.5.0 ({})
```

Comment:
You may want to be sure to run this test via `make test only=cargo_test_release` as the output has probably changed with the `-v` flag!